tests/plugins/range.test.ts

```diff
@@ -434,6 +434,13 @@ describe("range plugin", () => {
         expect(m.getters.getRangeString(range, "tao")).toBe(`'${name}'!A1`);
       }
     );
+
+    test.each([["s1!A1:s1!A9"], ["s1!!!A1:s1!!!A9"]])(
```

Comment:
I'm pretty sure we don't properly support everywhere ranges with the sheet name on both sides.
It require a bit more extensive testing than that.

Also `"s1!!!A1:s1!!!A9"` is not a valid string reference :wink:  If there is a `!` in the sheet name, then the sheet name needs to be enclosed with single quotes `"'s1!!'!A1:'s1!!'!A9"`

Reply:
> I'm pretty sure we don't properly support everywhere ranges with the sheet name on both sides. It requires a bit more extensive testing than that (does the tokenizer handle that correctly?).

I checked the range string which is split by "!" and the result arrays are all reversed, so it will not cause error so that no freezing or crashes. 

For the correctness of parsing multiple ! marks, I think none of them can do it correctly for now. For example, `s1!A1:s1!A9` will be split as `["s1", "A1:s1", "A9"]`. But because the input border will become red when inputs are like this, I think it's clear for me to change my input. 

> Also `"s1!!!A1:s1!!!A9"` is not a valid string reference wink If there is a `!` in the sheet name, then the sheet name needs to be enclosed with single quotes `"'s1!!'!A1:'s1!!'!A9"`

Because we parse our range string first using "!", it will also cause error if the split result array is not reversed, and the input border will also be red (implies it is not a valid input).

Reply:
@rrahir So the first thing I need to test is something more valid, like `s1!!!A1:A9`, even when they're not parsed correctly for now?

Reply:
yup, I created a task to address that other issue, because it's also impacting version 15.0